R/spread_with_duplicates.R

```diff
@@ -0,0 +1,41 @@
+#' Spread a key-value pair across multiple columns in presence of duplicates
+#' 
+#' @param .df A dataframe
+#' @param key,value Column names or positions
+#' @param fill If set, missing values will be replaced with this value. 
+#' @return A data.frame
+#' @examples 
+#' test1 <- data.frame(taxonKey = c(1, 1, 1, 1),	
+#'                 key = c("A", "B", "C", "C"),
+#'                 value = c("R", "S", "T", "X"),
+#'                 stringsAsFactors = FALSE)
+#' spread_with_duplicates(test1, key, value, by = "taxonKey")
+#' # with NAs
+#' test2 <- data.frame(taxonKey = c(1, 1, 1, 2),	
+#'                 key = c("A", "C", "C", "A"),
+#'                 value = c("R", "T", "X", "R"),
+#'                 stringsAsFactors = FALSE)
+#' spread_with_duplicates(test2, key, value, by = "taxonKey")
+#' spread_with_duplicates(test2, key, value, by = "taxonKey", fill = 0)
+#' @export
+#' @importFrom purrr map map2 reduce
+#' @importFrom dplyr mutate_all filter full_join pull
+#' @importFrom magrittr %>%
+#' @importFrom tidyselect vars_pull enquo
+spread_with_duplicates <- function(.df, key, value, by, fill = NA){
```

Comment:
out of curiosity: why `.df`? not just `df`?

Reply:
ahah, nice question! Actually it is because at the very beginning I was thinking to use `.data` as in `dplyr` package (they use `.data` to avoid collision with function `data()` from base R) but then I changed it to `.df` and the point is absolutely not useful. Probably because I use often `df` as variable in testing. Thanks for noticing it!
I changed back to `.data`. See https://github.com/trias-project/trias/pull/17/commits/1fb7beac4fee62d923b757581ed046334fcc4cf7
So, if we ever send this function to dplyr developers as suggestion to handle duplicates in `spread()`, at least the name of the variable is already as they like.